### ISSUE TITLE
"Stop editing"/"Edit" button doesn't work correctly in Mac version

### ISSUE THREAD

**Issue by edo-vi** [fabiospampinato/notable#133]
In the Mac version, clicking the top left part of the "edit"/"stop editing" button doesn't work, only the bottom right part

Mac version: Mojave 10.14.2 (18C54)
Notable Version: 1.0.1

**Comment by fabiospampinato** [comment]
Are you sure about the "bottom right part"? I've also encountered this problem but I think the entire button wasn't responding to clicks, right now the button is working for me and I can't see any difference between the "bottom right part" or any other part.

**Comment by mschmid** [comment]
Thats interesting, for me the lower half of the button works, the upper half doesn't. Mojave 10.14.1

**Comment by fabiospampinato** [comment]
@mschmid do the other buttons work normally? Can you reproduce this with a freshly started-up app?

**Comment by edo-vi** [comment]
> Are you sure about the "bottom right part"? I've also encountered this problem but I think the entire button wasn't responding to clicks, right now the button is working for me and I can't see any difference between the "bottom right part" or any other part.

Yes, the bottom right part works consistently. All other buttons behave correctly tho

**Comment by fabiospampinato** [comment]
@mscgmid weird, could you maybe make a screen recording of that?

**Comment by jinjie** [comment]
@fabiospampinato @mschmid I also realised this issue. I have recorded a screen for it:

https://www.dropbox.com/s/siyhe6q6trf4g8v/Notable%20133.mov?dl=0

**Comment by mschmid** [comment]
The screen shows the exact same behavior as on my iMac.
On more hint: On my Macbook Pro running 10.13.6 the button works as it should.

**Comment by fabiospampinato** [comment]
@jinjie very weird, thanks a lot for the recording!

**Comment by MoHD20** [comment]
For the record, It works just fine on my Mac version `10.12.3` Notable Version `1.0.1`

**Comment by linus345** [comment]
> Thats interesting, for me the lower half of the button works, the upper half doesn't. Mojave 10.14.1

The same thing happens for me with Mojave `10.14.2` and Notable version `1.0.1`.
One thing I noticed is that the bug behaves differently (the button works in different places ex. lower half and not upper half) when I change the size of the window. For example when I make the window small the whole button works to click on but when I make it bigger the upper half stops working.

Another example that also affects the "tag" button:
If I make the window as wide and as low as possible like this:
<img width="1440" alt="notable-bug" src="https://user-images.githubusercontent.com/46067241/50607758-9a837580-0eca-11e9-9786-b9879310ab66.png">
Then the lower half of the "edit" button works but not the upper half and the "tag" button doesn't work in the upper left corner.

**Comment by fabiospampinato** [comment]
I can reproduce this occasionally, and it really looks like a but with Electron itself perhaps, there's nothing in front of that button.

If anyone would like to help me debug this or find a workaround I'd appreciate that. Maybe this has been fixed in Electron v4 or something 🤔

**Comment by mschmid** [comment]
Just for information: This bug is still present in Notable V1.1.0

**Comment by shallinta** [comment]
I have the same problem with @linus345. Only the lower half of the "edit" button works, and the upper left corner of "tag" button doesn't work. 
My notable version is v1.1.0, on my macOS Mojave 10.14.2.

Is there any cues?

**Comment by kadaan** [comment]
@fabiospampinato Not sure, but feels like it is related to: https://github.com/electron/electron/issues/14529

**Comment by kadaan** [comment]
@fabiospampinato Tried what was suggested [here](https://github.com/electron/electron/issues/16418#issuecomment-456263303) and it fixes it on my Mac, with no obvious (to me) visual change.

**Comment by kadaan** [comment]
@fabiospampinato Created a [PR](https://github.com/fabiospampinato/notable/pull/316) for this fix.  I'm not able to test on anything other than OsX Mojave, so I'm not sure how this will impact other platforms.

**Comment by jinjie** [comment]
Just tested locally. It seem to work.

Not sure is it just me, but I see the hand pointer at the lower part of the buttons. See:

![pointer](https://user-images.githubusercontent.com/2721195/51519032-b4c1cb00-1e5a-11e9-9e28-84568fb54039.gif)

**Comment by kadaan** [comment]
@jinjie Good catch, I get the hand pointer as well..  Kinda annoying, but at least the button reliably works.

**Comment by kadaan** [comment]
@jinjie Looks like Notable is using the `cursor: pointer` for the buttons, but that isn't really in line with other apps on OsX.  Might be easiest to not use `cursor: pointer` on those (so they use the default arrow), along with the `titleBarStyle` change.  I'm not sure how to make that cursor change as the styles come from some other project.

**Comment by wAuner** [comment]
I've encountered the same problem with notable 1.3.0 and macOS 10.14.3

The entire top part of the edit button as well as the top left part of the tag button does not react when I click on it.

**Comment by garbonsai** [comment]
Just confirming this bug still exists in 1.4.0 on Mojave. Bottom 1/2 of buttons work, top 1/2 doesn't.

**Comment by fabiospampinato** [comment]
@garbonsai Yeah this is an Electron bug. I think it may get fixed in Electron v5 (which is currently in beta), let's wait a bit more.

**Comment by fabiospampinato** [comment]
It looks like Electron v5](https://github.com/electron/electron/releases/tag/v5.0.0) has just been released! Hopefully upgrading to it should be relatively painless.

**Comment by fabiospampinato** [comment]
I've just upgraded to Electron v5 and sadly the issue is still present. I say let's wait some more releases, if it still doesn't get fixed the only solution that doesn't sacrifice aesthetics would be basically re-implementing those buttons via CSS, which I'd prefer to avoid doing.

**Comment by fabiospampinato** [comment]
Note to future me: remember that the window semaphore is styled slightly differently when the window is not focused depending on if macOS's dark appearance is enabled or not.

**Comment by fabiospampinato** [comment]
Turns out there's a super ugly [workaround](https://github.com/electron/electron/issues/14529#issuecomment-426021665) for this, I've just implemented it and now the unresponsive area should be about 10 times smaller. 

This is not a proper workaround as it doesn't address the problem entirely, nor a proper fix which must be implemented in Electron, but at least the situation should be a bit better now.
